lib/saml_idp/service_provider.rb

```diff
@@ -21,13 +21,20 @@ def valid?
     end
 
     def valid_signature?(doc, require_signature = false, options = {})
-      Array(certs).any? do |cert|
+      @matching_cert = Array(certs).find do |cert|
         if doc.valid_signature?(fingerprint_cert(cert), options.merge(cert: cert))
-          @matching_cert = cert
+          cert
         end
```

Comment:
if we're switching to a `find`, the block just needs to return `true` so we can get rid of the `if` and just have that condition be the body

```ruby
      @matching_cert = Array(certs).find do |cert|
        doc.valid_signature?(fingerprint_cert(cert), options.merge(cert: cert))
      end
```

Reply:
updated in https://github.com/18F/saml_idp/pull/86/commits/0d555dad26c6001c1f214b6adbe68a33aa9fb56e